### PR TITLE
Add outpost_arn to instance datasource

### PR DESCRIPTION
**Pull request by johnbarney**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #12302

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
Adds Outpost support to instance datasource
Adds Outpost support to instance resource
```

Output from acceptance testing:

```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSInstanceDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSInstanceDataSource_basic -timeout 120m
=== RUN   TestAccAWSInstanceDataSource_basic
=== PAUSE TestAccAWSInstanceDataSource_basic
=== CONT  TestAccAWSInstanceDataSource_basic
--- PASS: TestAccAWSInstanceDataSource_basic (194.13s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	195.632s

$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSInstance_basic -timeout 120m
=== RUN   TestAccAWSInstance_basic
=== PAUSE TestAccAWSInstance_basic
=== CONT  TestAccAWSInstance_basic
--- PASS: TestAccAWSInstance_basic (311.96s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	313.473s
```

### CODE REVIEW

**Comment by johnbarney** [comment]
Requires #12097 to run test in addition to an Outpost to test with

**Comment by ewbankkit** [comment]
@johnbarney Thanks for this.
You could add a check to some (at least `TestAccAWSInstanceDataSource_basic`) of the other acceptance tests to check that the `outpost_arn` attribute is empty.

**Comment by johnbarney** [comment]
Done
```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSInstanceDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSInstanceDataSource_basic -timeout 120m
=== RUN   TestAccAWSInstanceDataSource_basic
=== PAUSE TestAccAWSInstanceDataSource_basic
=== CONT  TestAccAWSInstanceDataSource_basic
--- PASS: TestAccAWSInstanceDataSource_basic (161.09s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       162.630s
```

**Comment by johnbarney** [comment]
Requires changes from #12097

```
$ AWS_OUTPOST_ARN=arn:aws:outposts:us-west-2:<snip>:outpost/op-<snip> make testacc TEST=./aws TESTARGS='-run=TestAccAWSInstance_outpost'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSInstance_outpost -timeout 120m
=== RUN   TestAccAWSInstance_outpost
=== PAUSE TestAccAWSInstance_outpost
=== CONT  TestAccAWSInstance_outpost
--- PASS: TestAccAWSInstance_outpost (273.50s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	275.827s
```

File: aws/resource_aws_instance_test.go [hunk 1]
```diff
@@ -278,6 +278,7 @@ func TestAccAWSInstance_basic(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "user_data", "3dc39dda39be1205215e776bad998da361a5955d"),
 					resource.TestCheckResourceAttr(resourceName, "root_block_device.#", "0"), // This is an instance store AMI
 					resource.TestCheckResourceAttr(resourceName, "ebs_block_device.#", "0"),
+          resource.TestCheckResourceAttr(resourceName, "outpost_arn", ""),
```
Comment (bflad) [inline aws/resource_aws_instance_test.go:281]: Fix for CI error:

```suggestion
					resource.TestCheckResourceAttr(resourceName, "outpost_arn", ""),
```

**Pull request merged**